test/replace-state-patch.html

```diff
@@ -46,7 +48,7 @@
         var resolve;
         // this is for setups that return promises
         var cooledDown = new Promise(function (res) {
```

Comment:
Having cooledDown and resolve and res seems a bit overcomplicated. Why not just:

```javascript
window.cooldownFunction = function (done, force) {
  return new Promise(function (resolve) {
    // Rest of logic in here
  });
};
```

This means more indentation, but you can recover that by doing an early exit if you don't need to do the cooldown. So:

```javascript
var shouldCooldown = runnerWindow.replaceStateCount > 90 || force;
if (!shouldCooldown) {
  resolve();
  return;
}
var cooldownPeriod = 30 * 1000;
// etc
```